### ISSUE TITLE
DisconnectAsync hangs on Windows and Android after scanning then connecting with ConnectToKnownDeviceAsync

### ISSUE THREAD

**Issue by TravisRo** [dotnet-bluetooth-le/dotnet-bluetooth-le#734]
To help us fix your issue, please provide the information in the below template. If something causes a crash, provide as much information as you can gather.
Just imagine: we do not know what you are doing! 

> Note: There is often little we can do without a minimal reproducible sample of the issue, so please provide that in a **standalone git repository** and link it here.

## Steps to reproduce

1. Start device scanning

2. Stop device scanning after test device is discovered

3. Connect to test device using `ConnectToKnownDeviceAsync`

4. Disconnect from test device

5. Test device does not disconnect and `DisconnectDeviceAsync` hangs

## Expected behavior
Device should disconnect successfully

## Actual behavior
Device does not disconnect and `DisconnectDeviceAsync` hangs

### Crashlog

If something causes an exception paste full stack trace + Exception here

## Configuration

**Version of the Plugin:** 3.0.0-beta.5

**Platform:** Android 13/Windows 10

**Device:** Galaxy S23 Ultra/Windows 10

## Sample code
```CSharp
using System.Collections.ObjectModel;
using MetroLog;
using Plugin.BLE;
using Plugin.BLE.Abstractions;
using Plugin.BLE.Abstractions.Contracts;
using Plugin.BLE.Abstractions.EventArgs;

namespace MauBleTest
{
    public class DeviceItem
    {
        public string Name { get; set; }
        public IDevice Device { get; set; }
    }
    public class MainViewModel:TinyMvvm.TinyViewModel
    {
        internal static readonly Guid TTS_BLE_SERVICE_ID = Guid.Parse("{0000abf0-0000-1000-8000-00805f9b34fb}");
        internal static readonly Guid TTS_BLE_CHAR_DATA_RECEIVE_ID = Guid.Parse("{0000abf1-0000-1000-8000-00805f9b34fb}");
        internal static readonly Guid TTS_BLE_CHAR_DATA_NOTIFY_ID = Guid.Parse("{0000abf2-0000-1000-8000-00805f9b34fb}");
        internal static readonly Guid TTS_BLE_CHAR_COMMAND_RECEIVE_ID = Guid.Parse("{0000abf3-0000-1000-8000-00805f9b34fb}");
        internal static readonly Guid TTS_BLE_CHAR_COMMAND_NOTIFY_ID = Guid.Parse("{0000abf4-0000-1000-8000-00805f9b34fb}");

        public static MainViewModel Instance { get; }= new MainViewModel();
        private ObservableCollection<DeviceItem> _deviceList = new();
        private bool _isScanIdle = true;
        private bool _isConnected;
        private DeviceItem _selectedDevice;
        private readonly IAdapter adapter;
        private IDevice _dev;
        private string _statusText = "";

        public ILogger Log = LoggerFactory.GetLogger(nameof(MauBleTest));
        public MainViewModel()
        {
            adapter = CrossBluetoothLE.Current.Adapter;
            adapter.DeviceConnectionError += OnDeviceConnectionError;
            adapter.DeviceConnectionLost += OnDeviceConnectionLost;
            adapter.DeviceConnected += OnDeviceConnected;
            adapter.DeviceDisconnected += OnDeviceDisconnected;
            adapter.DeviceDiscovered += OnDeviceDiscovered;
        }

        private void OnDeviceDiscovered(object sender, DeviceEventArgs e)
        {
            if (!MainThread.IsMainThread)
            {
                MainThread.BeginInvokeOnMainThread(delegate { OnDeviceDiscovered(sender, e); });
                return;
            }

            DeviceList.Add(new DeviceItem() { Name = e.Device.Name, Device = e.Device });
        }
        private Command _menuCommand;
        public Command MenuCommand => _menuCommand ??= new Command<object>(OnMenuCommand);

        private void OnMenuCommand(object obj)
        {
            if (obj == null) return;
            switch (obj.ToString())
            {
                case "Scan":
                    startScan();
                    break;
                case "Connect":
                    connect();
                    break;
            }
        }

        public string StatusText
        {
            get => _statusText;
            set
            {
                if (value == _statusText) return;
                _statusText = value;
                OnPropertyChanged();
            }
        }

        private async void connect()
        {
            IPermissionsService service = DependencyService.Get<IPermissionsService>();
            if (service != null)
            {
                var blePerm = await service.RequestBlePermissions();
                if (blePerm != true) return;
            }

#if !WINDOWS
            var locPerm = await Permissions.CheckStatusAsync<Permissions.LocationWhenInUse>();
            if (locPerm != PermissionStatus.Granted)
            {
                locPerm = await Permissions.RequestAsync<Permissions.LocationWhenInUse>();
                if (locPerm != PermissionStatus.Granted) return;
            }
#endif
            if (IsConnected)
            {
                if (_dev != null)
                {
                    StatusText += "Disconnecting..";
                    await adapter.DisconnectDeviceAsync(_dev);

                }

                return;
            }
            if (SelectedDevice == null) return;

            StatusText+= "Connecting..";
            _dev = SelectedDevice.Device;
            try
            {
                await adapter.ConnectToKnownDeviceAsync(_dev.Id, connectParameters: new ConnectParameters(false, true));

            }
            catch (Exception ex)
            {
                StatusText += $"Connect failed. {ex.Message}";

            }
        }

        private void OnDeviceConnectionError(object sender, DeviceErrorEventArgs e)
        {
            StatusText += $"Device connection error. {e.ErrorMessage}.";
            Log.Error(e.ErrorMessage);
        }

        private void OnDeviceConnectionLost(object sender, DeviceErrorEventArgs e)
        {
            StatusText += $"Device connection lost. {e.ErrorMessage}.";
            Log.Error(e.ErrorMessage);
            IsConnected = false;
        }

        private void OnDeviceDisconnected(object sender, DeviceEventArgs e)
        {
            StatusText += "Disconnected!";
            IsConnected = false;
        }

        private void OnDeviceConnected(object sender, DeviceEventArgs e)
        {
            StatusText += "Connected!";
            IsConnected = true;
        }

        private async void startScan()
        {
            IPermissionsService service = DependencyService.Get<IPermissionsService>();
            if (service != null)
            {
                var blePerm = await service.RequestBlePermissions();
                if (blePerm != true) return;
            }

#if !WINDOWS
            var locPerm = await Permissions.CheckStatusAsync<Permissions.LocationWhenInUse>();
            if (locPerm != PermissionStatus.Granted)
            {
                locPerm = await Permissions.RequestAsync<Permissions.LocationWhenInUse>();
                if (locPerm != PermissionStatus.Granted) return;
            }
#endif
            if (IsScanIdle)
            {
                IsScanIdle = false;
                await adapter.StartScanningForDevicesAsync(serviceUuids: new[] { TTS_BLE_SERVICE_ID });

            }
            else
            {
                await adapter.StopScanningForDevicesAsync();
                IsScanIdle = true;
            }
        }
        public ObservableCollection<DeviceItem> DeviceList
        {
            get => _deviceList;
            set
            {
                if (Equals(value, _deviceList)) return;
                _deviceList = value;
                OnPropertyChanged();
            }
        }

        public bool IsScanIdle
        {
            get => _isScanIdle;
            set
            {
                if (Equals(value, _isScanIdle)) return;
                _isScanIdle = value;
                OnPropertyChanged();
            }
        }

        public bool IsConnected
        {
            get => _isConnected;
            set
            {
                if (value == _isConnected) return;
                _isConnected = value;
                OnPropertyChanged();
            }
        }

        public DeviceItem SelectedDevice
        {
            get => _selectedDevice;
            set
            {
                if (Equals(value, _selectedDevice)) return;
                _selectedDevice = value;
                OnPropertyChanged();
            }
        }
    }
}

```

**Comment by smsissuechecker** [comment]
Hi @TravisRo,

I'm the friendly issue checker.
Thanks for using the issue template :star2:
I appreciate it very much. I'm sure, the maintainers of this repository will answer, soon.

**Comment by janusw** [comment]
Fixed via #735. Closing.
